core/object/object.cpp

```diff
@@ -1100,6 +1100,20 @@ bool Object::_has_user_signal(const StringName &p_name) const {
 	return signal_map[p_name].user.name.length() > 0;
 }
 
+void Object::_remove_user_signal(const StringName &p_name) {
+	ERR_FAIL_COND_MSG(!signal_map.has(p_name), "Provided signal does not exist.");
+	SignalData *s = signal_map.getptr(p_name);
+	ERR_FAIL_COND_MSG(!s->removable, "Provided signal is not removable.");
```

Comment:
Maybe this should be more explicit on what can make make a signal not removable. maybe adding something like "as it was not added through `add_user_signal`" or something else that explicits it can't be removed because it is not a user sigal.

Reply:
Sure i'll clarify the error